models/aether_lat-lon/aether_to_dart.f90

```diff
@@ -0,0 +1,472 @@
+
+if (define) then
+   ! if define, run one block.
+   ! the block_to_filter_io call defines the variables in the whole domain netCDF file.
+   ib_loop = 1
+   jb_loop = 1
+   call nc_begin_define_mode(ncid_output)
+else
+   ! if not define, run all blocks.
+   ! the block_to_filter_io call adds the (ib,jb) block to the netCDF variables 
+   ! in order to make a file containing the data for all the blocks.
+   ib_loop = nblocks_lon
+   jb_loop = nblocks_lat
+end if
+
+do jb = 1, jb_loop
+   do ib = 1, ib_loop
+      call block_to_filter_io(ncid_output, ib, jb, member, define)
+   enddo
+enddo
+
+if (define) then
+   call nc_end_define_mode(ncid_output)
+endif
+    
+end subroutine restarts_to_filter
+
+!-----------------------------------------------------------------------
+! Transfer variable data from a block restart file to the filter_input.nc file. 
+! It's called with 2 modes:
+! define = .true.  define the NC variables in the filter_input.nc 
+! define = .false. write the data from a block to the NC file using write_filter_io.
+
+subroutine block_to_filter_io(ncid_output, ib, jb, member, define)
+
+integer,          intent(in) :: ncid_output
+integer,          intent(in) :: ib, jb
+integer,          intent(in) :: member
+logical,          intent(in) :: define
+
+real(r4), allocatable :: temp1d(:), temp2d(:,:), temp3d(:,:,:)
+! real(r4), allocatable :: alt1d(:), density_ion_e(:,:,:)
+integer               :: ivar, nb, ncid_input
+! TEC? integer               :: maxsize
+!      logical               :: no_idensity
+!      real(r4)              :: temp0d 
+character(len=32)     :: att_val 
+character(len=128)    :: file_root 
+character(len=256)    :: filename
+character(len=vtablenamelength) :: varname, dart_varname
+
+character(len=*), parameter :: routine = 'block_to_filter_io'
+
+! The block number, as counted in Aether.
+! Lower left is 0, increase to the East, then 1 row farther north, West to East.
+nb = (jb - 1) * nblocks_lon + ib - 1
+
+! a temp array large enough to hold any of the
+! Lon,Lat or Alt array from a block plus ghost cells
+allocate(temp1d(1-nghost:max(nx_per_block, ny_per_block, nz_per_block) + nghost))
+
+! treat alt specially since we want to derive TEC here
+! TODO: See density_ion_e too.
+! allocate( alt1d(1-nghost:max(nx_per_block, ny_per_block, nz_per_block) + nghost))
+
+! temp array large enough to hold any 2D field 
+allocate(temp2d(1-nghost:ny_per_block+nghost, &
+                1-nghost:nx_per_block+nghost))
+
+! TODO: We need all altitudes, but there might be vertical blocks in the future.
+!       But there would be no vertical halos.
+!       Make transform_state_mod: zcount adapt to whether there are blocks.
+!       Temp needs to have C-ordering, which is what the restart files have.
+! temp array large enough to hold 1 species, temperature, etc
+allocate(temp3d(1:nz_per_block, &
+                1-nghost:ny_per_block+nghost, &
+                1-nghost:nx_per_block+nghost))
```

Comment:
I tried to ncview individual members are running aether_to_dart but the fields don't show up as I expect. I was hoping to see a global field and then change through the levels. However, I could only change through the lon dimension which is not very useful. This is because the dimensions of the 3D variables are ordered as lon, lat, alt. I ran a simple 'ncpdq -a alt,lon filter_input_0001.nc filter_input_0001_swap.nc' on the file to permute the dimensions and then I was able to see the entire globe at different levels. I suggest modifying aether_to_dart to adjust the dimension order.

Reply:
The coordinate order is annoying.  Aether won't change, and we decided to use it in order to make the conversion programs simpler to code and run faster (no transposing).  
Ncview can show a lat-lon slice by changing the axes.  But the lat and lon are transposed, so it's not perfect.  Other common viewing tools can transpose it.